static/js/message_list_view.js

```diff
@@ -369,6 +369,17 @@ MessageListView.prototype = {
                 });
             }
 
+            row.find('.zulip-katex').each(function () {
+                var disp = $(this).hasClass('zulip-katex-display');
+                try {
+                    katex.render(this.textContent, this,
+                                 {throwOnError: false, displayMode: disp});
+                } catch (ex) {
+                    blueslip.error(ex);
+                }
+                $(this).removeClass('zulip-katex');
+            });
```

Comment:
I think we need to move this earlier in the rendering path, before the messages are inserted into the DOM (possibly as early as `message_store.add_message_metadata`).  I think if we did that, then we could eliminate the flicker from text to TeX that currently happens when you reload the page after sending several messages with TeX in them.